src/Dialogue.js

```diff
@@ -26,42 +35,77 @@ export default class Dialogue {
     if (this.isComplete()) {
       return this.completeText;
     } else if (this.isBattle()) {
-      return `${this.conversation[this.progress].description}\n${this.startBattle()}`;
+      return `${this.activeText.description}\n${this.startBattle()}`;
     }
-    return this.conversation[this.progress].description;
+    return this.activeText.description;
   }
 
   /**
    * Restart the dialogue!
    */
   restart() {
-    this.progress = 0;
+    for (var text in this.conversation) {
+      this.activeText = text;
+      break;
+    }
   }
 
   /**
    * For help text, either return its current helpText or its completed text
    * (if complete)
    */
   help() {
-    return this.isComplete() ? this.completeHelp : this.conversation[this.progress].helpText;
+    return this.isComplete() ? this.completeHelp : this.activeText.helpText;
   }
 
   /**
-   * When executing the user's input, check against all possible choices and
-   * return true or false depending on if we have received a match.
+   * Progress the current text state and any associated battle.
+   *
+   * @return - true if there's nothing left to execute, false otherwise.
    */
   execute(input, respond, player) {
     if (this.isBattle()) {
       return this._handleBattle(input, respond, player);
     }
-    return createRegex(this.conversation[this.progress].aliases, false).test(input);
+    return true;//createRegex(this.activeText.aliases, false).test(input);
   }
 
   /**
+   * Given user input, evaluate and return the key of the next text state
+   * NOTE: If the user input matches multiple regexes,the first one in the list will be used
+   * 
+   * @input - string that prompted the conversation progression
+   * @return - Key name of the next text state in this.conversation, or false if
+   *            no appropriate state is found
+   */
+  getNextTextKey(input) {
+    console.log("getting next text key");
+    var progressionMap = this.activeText.progression;
+    console.log(progressionMap);
+    for(var key in progressionMap) {
+
+      // Ignore if the property is from prototype
+      if(!progressionMap.hasOwnProperty(key)) {
+        continue;
+      }
+
+      // Make a regex from the progression map entry and check user input against it
+      if (createRegex(progressionMap[key]).test(input)) {
+        console.log("found a match");
+        console.log(key);
+        return key;
+      }
+    }
+    console.log("no match for input");
+    console.log(input);
+    return false;
+  }
```

Comment:
depending on what editor you're using they're might be a plugin for eslint - i use atom and every time i hit save it runs the lint on that file and let's you know in line what's going down